models/ed/R/write.configs.ed.R

```diff
@@ -382,12 +382,15 @@ write.config.xml.ED2 <- function(settings, trait.values, defaults=settings$const
 
       ## Convert trait values to ED units
       converted.trait.values <- convert.samples.ED(trait.values[[i]])
-
+      
       ## Selectively replace defaults with trait values
       vals <- modifyList(vals, converted.trait.values)
 
+      ## Convert settings constants to ED units
+      converted.defaults <- convert.samples.ED(defaults[[pft]])
+
       ## Selectively replace defaults and trait values with constants from settings
-      if(!is.null(defaults)) vals <- modifyList(vals, defaults)
+      if(!is.null(defaults)) vals <- modifyList(vals, converted.defaults)
```

Comment:
@ashiklom this line broke the code, because something is still not right the way things are passed to the defaults (I think it is related to getting rid of this [line](https://github.com/PecanProject/pecan/blob/master/models/ed/R/write.configs.ed.R#L158). see my other comment above). As a result defaults[[pft]] above is empty and converted.defaults is also empty and therefore not a list which modifyList isn't happy about